--- libtorrent-libtorrent-1_0_7/src/bt_peer_connection.cpp	2015-11-14 09:10:59.000000000 +0200
+++ libtorrent_original/src/bt_peer_connection.cpp	2015-11-14 11:12:06.000000000 +0200
@@ -3278,8 +3278,8 @@
 			{
 				m_statistics.received_bytes(0, bytes_transferred);
 				// packet too large
-				disconnect(errors::packet_too_large, 2);
-				return;
+				//disconnect(errors::packet_too_large, 2);
+				//return;
 			}
 					
 			if (packet_size == 0)
